src/plugins/intel_gna/src/gna_device.cpp

```diff
@@ -45,6 +45,8 @@ GNADeviceHelper::GNADeviceHelper(std::string executionTargetIn,
       isPerformanceMeasuring(isPerformanceMeasuring),
       nGnaDeviceIndex{selectGnaDevice()},
       useDeviceEmbeddedExport(deviceEmbedded) {
+    per_request_diagnostics = log::get_log_level() >= ov::log::Level::TRACE;
```

Comment:
I just wonder if we could have some common functionality for tracer/diagnostic based on config file, or arguments to GNAConfig which we relay not only on log levels but also on e.g. so call context (e.g. [Passes],[Backend],[Plugin],[Compiler]). So we could pass also list of context we would like to see if we would be not interested in all logs.

This would allow us to control logging information form on runtime not on compilation time.

Reply:
Good idea.